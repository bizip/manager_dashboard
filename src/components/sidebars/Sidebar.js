import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import { BsPieChart } from 'react-icons/bs';
import { HiOutlinePresentationChartLine } from 'react-icons/hi';
import { TbChartPie4 } from 'react-icons/tb';
import { MdOutlineSettingsApplications } from 'react-icons/md';
import { SlDiamond } from 'react-icons/sl';
import { AiOutlineTable } from 'react-icons/ai';
import { IoIosArrowDown } from 'react-icons/io';
import SubElement from '../subNavBars/SubElement';

const Sidebar = () => {
  const [element, setElement] = useState(false);

  const toggleElement = () => {
    setElement((prev) => !prev);
  };
  return (
    <section className="sidebar">
      <h3 className="side_header">UI COMPONENTS</h3>
      <ul className="elements">
        <li className="link_container">
          <div className="link_left">
            <SlDiamond className="link__icon" />
            <NavLink to="." className="link__sidebar" onClick={toggleElement}>Elements</NavLink>
          </div>
          <IoIosArrowDown className="link__icon" />
        </li>
        <SubElement isElement={element} />
        <li className="link_container">
          <div className="link_left">
            <div className="link_left">
              <HiOutlinePresentationChartLine className="link__icon" />
              <NavLink to="example2" activeClassName="active" className="link__sidebar">Components</NavLink>
            </div>
          </div>
          <IoIosArrowDown className="link__icon" />
        </li>
        <li className="link_container">
          <div className="link_left">
            <AiOutlineTable className="link__icon" />
            <NavLink to="example3" activeClassName="active" className="link__sidebar">Tables</NavLink>
          </div>
          <IoIosArrowDown className="link__icon" />
        </li>
      </ul>

      <h3 className="side_header">DASHBOARD WIDGETS</h3>
      <ul className="widgets">
        <li className="link_container">
          <div className="link_left">
            <BsPieChart className="link__icon" />
            <NavLink to="example4" activeClassName="active" className="link__sidebar">Chart boxes 1</NavLink>
          </div>
        </li>
        <li className="link_container">
          <div className="link_left">
            <HiOutlinePresentationChartLine className="link__icon" />
            <NavLink to="example5" activeClassName="active" className="link__sidebar">Chart boxes 2</NavLink>
          </div>
        </li>
        <li className="link_container">
          <div className="link_left">
            <TbChartPie4 className="link__icon" />
            <NavLink to="example6" activeClassName="active" className="link__sidebar">Chart boxes 3</NavLink>
          </div>
        </li>
        <li className="link_container">
          <div className="link_left">
            <MdOutlineSettingsApplications className="link__icon" />
            <NavLink to="exmple7" activeClassName="active" className="link__sidebar">Profile Boxes</NavLink>
          </div>
        </li>
        <li className="link_container">
          <div className="link_left">

            <MdOutlineSettingsApplications className="link__icon" />
            <NavLink to="example8" activeClassName="active" className="link__sidebar">Applications</NavLink>
          </div>
        </li>
      </ul>
    </section>
  );
};

export default Sidebar;
