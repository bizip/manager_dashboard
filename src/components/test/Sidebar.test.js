import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import Sidebar from '../sidebars/Sidebar';

describe('Test for rendering dashboard page', () => {
  it('It should render dashboard page', () => {
    const sidebar = renderer
      .create(<Router><Sidebar /></Router>)
      .toJSON();
    expect(sidebar).toMatchSnapshot();
  });
});
