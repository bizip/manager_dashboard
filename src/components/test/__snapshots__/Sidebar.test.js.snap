// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for rendering dashboard page It should render dashboard page 1`] = `
<section
  className="sidebar"
>
  <h3
    className="side_header"
  >
    UI COMPONENTS
  </h3>
  <ul
    className="elements"
  >
    <li
      className="link_container"
    >
      <div
        className="link_left"
      >
        <svg
          className="link__icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1018.72 295.472L878.848 28.03C870.688 12.43 849.584-.322 832-.322H510.064c-.351-.015-.703-.11-1.054-.127-.288 0-.56.113-.849.128h-316.16c-17.6 0-38.752 12.72-47.024 28.256L5.473 290.223c-8.289 15.536-6.624 39.937 3.631 54.257l480.016 669.152c5.153 7.184 12 10.815 18.832 10.815 6.785 0 13.584-3.536 18.768-10.591L1014.624 349.6c10.384-14.193 12.256-38.544 4.096-54.128zm-76.353-7.843H770.911l68.656-196.608zM575.343 63.677h205.968l-63.888 182.928zm92.895 223.952H370.591L511.263 85.533zm-354.351-30.544L249.71 63.677h198.816zm366.863 94.544L508.718 844.173 345.262 351.629H680.75zM436.926 831.085L92.99 351.629h184.832zm311.616-479.456H933.71l-352.976 480.56zM188.478 82.413l68.096 205.216H79.326z"
          />
        </svg>
        <a
          className="link__didebar"
          href="#/all"
          onClick={[Function]}
        >
          Elements
        </a>
      </div>
      <svg
        className="link__icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 294.1L383 167c9.4-9.4 24.6-9.4 33.9 0s9.3 24.6 0 34L273 345c-9.1 9.1-23.7 9.3-33.1.7L95 201.1c-4.7-4.7-7-10.9-7-17s2.3-12.3 7-17c9.4-9.4 24.6-9.4 33.9 0l127.1 127z"
        />
      </svg>
    </li>
    <li
      className="link_container"
    >
      <div
        className="link_left"
      >
        <div
          className="link_left"
        >
          <svg
            className="link__icon"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 12l3-3 3 3 4-4M8 21l4-4 4 4M3 4h18M4 4h16v12a1 1 0 01-1 1H5a1 1 0 01-1-1V4z"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
          <a
            activeClassName="active"
            className="link__didebar"
            href="#/example2"
            onClick={[Function]}
          >
            Components
          </a>
        </div>
      </div>
      <svg
        className="link__icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 294.1L383 167c9.4-9.4 24.6-9.4 33.9 0s9.3 24.6 0 34L273 345c-9.1 9.1-23.7 9.3-33.1.7L95 201.1c-4.7-4.7-7-10.9-7-17s2.3-12.3 7-17c9.4-9.4 24.6-9.4 33.9 0l127.1 127z"
        />
      </svg>
    </li>
    <li
      className="link_container"
    >
      <div
        className="link_left"
      >
        <svg
          className="link__icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 208H676V232h212v136zm0 224H676V432h212v160zM412 432h200v160H412V432zm200-64H412V232h200v136zm-476 64h212v160H136V432zm0-200h212v136H136V232zm0 424h212v136H136V656zm276 0h200v136H412V656zm476 136H676V656h212v136z"
          />
        </svg>
        <a
          activeClassName="active"
          className="link__didebar"
          href="#/example3"
          onClick={[Function]}
        >
          Tables
        </a>
      </div>
      <svg
        className="link__icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 294.1L383 167c9.4-9.4 24.6-9.4 33.9 0s9.3 24.6 0 34L273 345c-9.1 9.1-23.7 9.3-33.1.7L95 201.1c-4.7-4.7-7-10.9-7-17s2.3-12.3 7-17c9.4-9.4 24.6-9.4 33.9 0l127.1 127z"
        />
      </svg>
    </li>
  </ul>
  <h3
    className="side_header"
  >
    DASHBOARD WIDGETS
  </h3>
  <ul
    className="widgets"
  >
    <li
      className="link_container"
    >
      <div
        className="link_left"
      >
        <svg
          className="link__icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.5 1.018a7 7 0 0 0-4.79 11.566L7.5 7.793V1.018zm1 0V7.5h6.482A7.001 7.001 0 0 0 8.5 1.018zM14.982 8.5H8.207l-4.79 4.79A7 7 0 0 0 14.982 8.5zM0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8z"
          />
        </svg>
        <a
          activeClassName="active"
          className="link__didebar"
          href="#/example4"
          onClick={[Function]}
        >
          Chart boxes 1
        </a>
      </div>
    </li>
    <li
      className="link_container"
    >
      <div
        className="link_left"
      >
        <svg
          className="link__icon"
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 12l3-3 3 3 4-4M8 21l4-4 4 4M3 4h18M4 4h16v12a1 1 0 01-1 1H5a1 1 0 01-1-1V4z"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
        </svg>
        <a
          activeClassName="active"
          className="link__didebar"
          href="#/example5"
          onClick={[Function]}
        >
          Chart boxes 2
        </a>
      </div>
    </li>
    <li
      className="link_container"
    >
      <div
        className="link_left"
      >
        <svg
          className="link__icon"
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <desc />
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <path
            d="M12 12l-6.5 5.5"
          />
          <path
            d="M12 3v9h9"
          />
          <circle
            cx="12"
            cy="12"
            r="9"
          />
          <path
            d="M12 12l5 7.5"
          />
        </svg>
        <a
          activeClassName="active"
          className="link__didebar"
          href="#/example6"
          onClick={[Function]}
        >
          Chart boxes 3
        </a>
      </div>
    </li>
    <li
      className="link_container"
    >
      <div
        className="link_left"
      >
        <svg
          className="link__icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M6.21 13.97l1.2 2.07c.08.13.23.18.37.13l1.49-.6c.31.24.64.44 1.01.59l.22 1.59c.03.14.15.25.3.25h2.4c.15 0 .27-.11.3-.26l.22-1.59c.36-.15.7-.35 1.01-.59l1.49.6c.14.05.29 0 .37-.13l1.2-2.07c.08-.13.04-.29-.07-.39l-1.27-.99c.03-.19.04-.39.04-.58 0-.2-.02-.39-.04-.59l1.27-.99c.11-.09.15-.26.07-.39l-1.2-2.07a.306.306 0 00-.37-.13l-1.49.6c-.31-.24-.64-.44-1.01-.59l-.22-1.59a.31.31 0 00-.3-.25h-2.4a.31.31 0 00-.3.26l-.22 1.59c-.36.15-.71.34-1.01.58l-1.49-.6c-.14-.05-.29 0-.37.13l-1.2 2.07c-.08.13-.04.29.07.39l1.27.99c-.03.2-.05.39-.05.59 0 .2.02.39.04.59l-1.27.99c-.11.1-.14.26-.06.39zM12 10.29c.94 0 1.71.77 1.71 1.71s-.77 1.71-1.71 1.71-1.71-.77-1.71-1.71.77-1.71 1.71-1.71zM19 3H5a2 2 0 00-2 2v14a2 2 0 002 2h14a2 2 0 002-2V5a2 2 0 00-2-2zm0 16H5V5h14v14z"
          />
        </svg>
        <a
          activeClassName="active"
          className="link__didebar"
          href="#/exmple7"
          onClick={[Function]}
        >
          Profile Boxes
        </a>
      </div>
    </li>
    <li
      className="link_container"
    >
      <div
        className="link_left"
      >
        <svg
          className="link__icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M6.21 13.97l1.2 2.07c.08.13.23.18.37.13l1.49-.6c.31.24.64.44 1.01.59l.22 1.59c.03.14.15.25.3.25h2.4c.15 0 .27-.11.3-.26l.22-1.59c.36-.15.7-.35 1.01-.59l1.49.6c.14.05.29 0 .37-.13l1.2-2.07c.08-.13.04-.29-.07-.39l-1.27-.99c.03-.19.04-.39.04-.58 0-.2-.02-.39-.04-.59l1.27-.99c.11-.09.15-.26.07-.39l-1.2-2.07a.306.306 0 00-.37-.13l-1.49.6c-.31-.24-.64-.44-1.01-.59l-.22-1.59a.31.31 0 00-.3-.25h-2.4a.31.31 0 00-.3.26l-.22 1.59c-.36.15-.71.34-1.01.58l-1.49-.6c-.14-.05-.29 0-.37.13l-1.2 2.07c-.08.13-.04.29.07.39l1.27.99c-.03.2-.05.39-.05.59 0 .2.02.39.04.59l-1.27.99c-.11.1-.14.26-.06.39zM12 10.29c.94 0 1.71.77 1.71 1.71s-.77 1.71-1.71 1.71-1.71-.77-1.71-1.71.77-1.71 1.71-1.71zM19 3H5a2 2 0 00-2 2v14a2 2 0 002 2h14a2 2 0 002-2V5a2 2 0 00-2-2zm0 16H5V5h14v14z"
          />
        </svg>
        <a
          activeClassName="active"
          className="link__didebar"
          href="#/example8"
          onClick={[Function]}
        >
          Applications
        </a>
      </div>
    </li>
  </ul>
</section>
`;
